python/python/tests/test_vector_index.py

```diff
@@ -541,33 +541,45 @@ def test_knn_with_deletions(tmp_path):
 def test_index_cache_size(tmp_path):
     rng = np.random.default_rng(seed=42)
 
-    def query_index(ds, ntimes):
```

Comment:
Had to make these changes because appearly calling `create_plan` hits the index cache somehow?

This test used to hard code hit rations, now the ratios are all relative to be more robust